src/bitmap/iter.rs

```diff
@@ -59,6 +80,41 @@ impl<'a> Drop for BitmapIterator<'a> {
     }
 }
 
+pub struct BitmapChunks<'a> {
+    iterator: BitmapIterator<'a>,
+    size: usize,
+}
+
+impl<'a> BitmapChunks<'a> {
+    pub fn new(iterator: BitmapIterator<'a>, size: usize) -> Self {
+        BitmapChunks { iterator, size }
+    }
+}
+
+impl<'a> Iterator for BitmapChunks<'a> {
+    type Item = Vec<u32>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let mut buffer = Vec::with_capacity(self.size);
```

Comment:
could this buffer be hoisted to live in the iterator struct? It would avoid new allocations for each chunk

Reply:
True, I see what you mean. Unfortunately I'm not sure how to make that happen. I see three ways:

1) Keep it on `struct` and return a copy. Still, an allocation happens.
2) Keep in on `struct` and return a slice to this vector on check iterator advancement. This sounds like a `StreamingIterator` problem described [here](http://lukaskalbertodt.github.io/2018/08/03/solving-the-generalized-streaming-iterator-problem-without-gats.html) and other places
3) Allow `.chunks()` to take a user supplied buffer, but not sure that's possible and idiomatic as well.

I'll try to dig into 2nd option, but can't make any promises, this looks tough.

Reply:
I _think_ that `type Item = &'a [u32]` could work, but I'm not sure.